Document the current state of the project in the readme (#1069)
whitequark/parser: README.md

Sponsored by [Evil Martians](http://evilmartians.com).
MacRuby and RubyMotion support sponsored by [CodeClimate](http://codeclimate.com).

> [!WARNING]  

> > The `parser` gem can only correctly parse Ruby code up to and including Ruby 3.3. For Ruby 3.4 and later, please use the [Prism translation parser](https://github.com/ruby/prism/blob/main/docs/parser_translation.md) instead.

Reviewer: That's just my two cents.

```suggestion
> The `parser` gem is only compatible parse Ruby code up to and including Ruby 3.3. For Ruby 3.4 and later, please use the [`Prism::Translation::Parser`](https://github.com/ruby/prism/blob/main/docs/parser_translation.md) instead.
```

Author: I tweaked the first part a bit since a word was missing otherwise

> > Starting in Ruby 3.4, Prism is the parser used in Ruby itself and can produces AST that is identical to the output of `parser`. If you only need to parse Ruby 3.3 (or greater), also consider using the native Prism AST.

Reviewer: It might be helpful to add a note clarifying the differences in compatibility between the respective ASTs.

```suggestion
> Starting in Ruby 3.4, Prism is the parser used in Ruby itself and can produces AST that is identical to the output of `parser`. If you only need to parse Ruby 3.3 (or greater) and don't require compatibility with the `parser` gem AST, also consider using the native Prism AST.
```
